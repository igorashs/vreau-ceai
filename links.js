import CartSvg from '@icons/cart.svg';
import LoginSvg from '@icons/login.svg';
import LogoutSvg from '@icons/logout.svg';
import PhoneSvg from '@icons/phone.svg';
import EmailSvg from '@icons/email.svg';
import PackageSvg from '@icons/package.svg';
import BriefcaseSvg from '@icons/briefcase.svg';

const linkFactory = ({
  path,
  text,
  label,
  Icon = null,
  accent = true,
  privatePath = false,
  managementPath = false
}) => ({
  path,
  text,
  label,
  Icon,
  accent,
  privatePath,
  managementPath
});

export const categoriesLink = linkFactory({
  path: '/categories',
  text: 'Ceai',
  label: 'Deschideți pagina cu categorii',
  accent: false
});

export const aboutLink = linkFactory({
  path: '/about',
  text: 'Despre',
  label: 'Deschideți pagina cu informații despre noi',
  accent: false
});

export const contactsLink = linkFactory({
  path: '/contacts',
  text: 'Contacte',
  label: 'Deschideți pagina cu contacte',
  accent: false
});

export const cartLink = linkFactory({
  path: '/cart',
  text: 'Coș',
  label: 'Deschideți pagina cu coșul personal',
  Icon: CartSvg,
  accent: false
});

export const loginLink = linkFactory({
  path: '/login',
  text: 'Conectați-vă',
  label: 'Deschideți pagina pentru conectare',
  Icon: LoginSvg
});

export const myOrdersLink = linkFactory({
  path: '/my-orders',
  text: 'Comenzile Mele',
  label: 'Deschideți pagina cu comenziile efectuate',
  Icon: PackageSvg,
  accent: false,
  privatePath: true
});

export const logoutLink = linkFactory({
  path: '/login',
  text: 'Deconectare',
  label: 'Deconectare din account',
  Icon: LogoutSvg,
  privatePath: true,
  accent: false
});

export const managementLink = linkFactory({
  path: '/management',
  text: 'Managment',
  label: 'Managment',
  Icon: BriefcaseSvg,
  privatePath: true,
  accent: true,
  managementPath: true
});

export const navLinks = [
  managementLink,
  categoriesLink,
  aboutLink,
  contactsLink,
  cartLink,
  loginLink,
  myOrdersLink,
  logoutLink
];

export const footerBlockLinksList = [
  {
    ...aboutLink,
    links: [
      linkFactory({
        path: '/about',
        text: 'vreau ceai',
        label: 'Deschideți pagina cu informații despre noi'
      })
    ]
  },
  {
    ...linkFactory({
      path: '/faq',
      text: 'FAQ',
      label: 'Deschideți pagina cu întrebări frecvente',
      accent: false
    }),
    links: [
      linkFactory({
        path: '/faq#cum-comand',
        text: 'cum comand',
        label: 'Deschideți pagina cu întrebarea cum comand'
      }),
      linkFactory({
        path: '/faq#cum-achit',
        text: 'cum achit',
        label: 'Deschideți pagina cu întrebarea cum achit'
      }),
      linkFactory({
        path: '/faq#livrare',
        text: 'cum are loc livrarea',
        label: 'Deschideți pagina cu întrebarea despre cum are loc livrarea'
      })
    ]
  },
  {
    ...contactsLink,
    links: [
      linkFactory({
        path: 'tel:062222222',
        text: '062222222',
        label: 'apelați nr de tel',
        Icon: PhoneSvg
      }),
      linkFactory({
        path: 'mailto:vreauceai@gmail.com',
        text: 'vreauceai@gmail.com',
        label: 'trimiteți email',
        Icon: EmailSvg
      })
    ]
  }
];
