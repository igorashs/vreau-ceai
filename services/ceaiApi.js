const URL = process.env.NEXT_PUBLIC_API_URL;

export const login = async (data) => {
  const res = await fetch(`${URL}/api/users/login`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json'
    },
    body: JSON.stringify(data)
  });

  const resData = await res.json();

  return resData;
};

export const signup = async (data) => {
  const res = await fetch(`${URL}/api/users/signup`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json'
    },
    body: JSON.stringify(data)
  });

  const resData = await res.json();

  return resData;
};

export const logout = async () => {
  const res = await fetch(`${URL}/api/users/logout`, {
    method: 'POST'
  });

  const resData = await res.json();

  return resData;
};

export const findUser = async (email) => {
  const res = await fetch(`${URL}/api/users?search=${email}`);

  const resData = await res.json();

  return resData;
};

export const updateUserManagerPermission = async (id, isManager) => {
  const res = await fetch(`${URL}/api/users/${id}`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json'
    },
    body: JSON.stringify({ isManager })
  });

  const resData = await res.json();

  return resData;
};

export const refreshUser = async () => {
  const res = await fetch(`${URL}/api/users/refresh`, {
    method: 'POST',
    credentials: 'same-origin'
  });

  const resData = await res.json();

  return resData;
};

export const createCategory = async (data) => {
  const res = await fetch(`${URL}/api/categories`, {
    method: 'POST',
    credentials: 'same-origin',
    headers: {
      'Content-type': 'application/json'
    },
    body: JSON.stringify(data)
  });

  const resData = await res.json();

  return resData;
};

export const getCategories = async () => {
  const res = await fetch(`${URL}/api/categories`);

  const resData = await res.json();

  return resData;
};

export const findCategory = async (name) => {
  const res = await fetch(`${URL}/api/categories?search=${name}`);

  const resData = await res.json();

  return resData;
};

export const updateCategory = async (id, data) => {
  const res = await fetch(`${URL}/api/categories/${id}`, {
    method: 'POST',
    credentials: 'same-origin',
    headers: {
      'Content-type': 'application/json'
    },
    body: JSON.stringify(data)
  });

  const resData = await res.json();

  return resData;
};

export const deleteCategory = async (id) => {
  const res = await fetch(`${URL}/api/categories/${id}`, {
    method: 'DELETE',
    credentials: 'same-origin'
  });

  const resData = await res.json();

  return resData;
};
