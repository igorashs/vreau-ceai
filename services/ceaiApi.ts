import {
  Category,
  ApiResponse,
  Product,
  UserLogin,
  UserSignup,
  UserEmail,
  User,
  UserPermissions,
  OrderFields,
  UserOrders,
  ProductWithCategory,
  ProductErrorDetail,
  ProductName,
  ManagementOrders,
  OrderStatus,
  Order,
  OrderStatusErrorDetail,
  OrderNumber,
  OrderNumberErrorDetail,
  UserEmailErrorDetail,
  ProductNameErrorDetail,
  CategoryName,
  CategoryNameErrorDetail,
  SessionAuth,
} from 'types';

const URL = process.env.NEXT_PUBLIC_API_URL;

export const login = async <P extends UserLogin, R extends ApiResponse>(
  data: P,
): Promise<R> => {
  const res = await fetch(`${URL}/api/users/login`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(data),
  });

  return res.json() as Promise<R>;
};

export const signup = async <P extends UserSignup, R extends ApiResponse>(
  data: P,
) => {
  const res = await fetch(`${URL}/api/users/signup`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(data),
  });

  return res.json() as Promise<R>;
};

export const logout = async <R extends ApiResponse>() => {
  const res = await fetch(`${URL}/api/users/logout`, {
    method: 'POST',
  });

  return res.json() as Promise<R>;
};

export const findUser = async <
  P extends UserEmail,
  R extends ApiResponse & { user: User; errors?: UserEmailErrorDetail[] }
>({
  email,
}: P) => {
  const res = await fetch(`${URL}/api/users?search=${email}`);

  return res.json() as Promise<R>;
};

export const updateUserManagerPermission = async <
  P extends UserPermissions,
  R extends ApiResponse & { user: User }
>(
  id: string,
  data: P,
) => {
  const res = await fetch(`${URL}/api/users/${id}`, {
    method: 'PUT',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(data),
  });

  return res.json() as Promise<R>;
};

export const refreshUser = async <
  R extends ApiResponse & { session: SessionAuth }
>() => {
  const res = await fetch(`${URL}/api/users/refresh`, {
    method: 'POST',
    credentials: 'same-origin',
  });

  return res.json() as Promise<R>;
};

export const createCategory = async <
  P extends CategoryName,
  R extends ApiResponse & {
    category: Category;
    errors?: CategoryNameErrorDetail[];
  }
>(
  data: P,
) => {
  const res = await fetch(`${URL}/api/categories`, {
    method: 'POST',
    credentials: 'same-origin',
    headers: {
      'Content-type': 'application/json',
    },
    body: JSON.stringify(data),
  });

  return res.json() as Promise<R>;
};

export const getCategories = async <
  R extends ApiResponse & { categories: Category[] }
>() => {
  const res = await fetch(`${URL}/api/categories`);

  return res.json() as Promise<R>;
};

export const findCategory = async <
  P extends CategoryName,
  R extends ApiResponse & {
    category: Category;
    errors?: CategoryNameErrorDetail[];
  }
>({
  name,
}: P) => {
  const res = await fetch(`${URL}/api/categories?search=${name}`);

  return res.json() as Promise<R>;
};

export const updateCategory = async <
  P extends CategoryName,
  R extends ApiResponse & {
    category: Category;
    errors?: CategoryNameErrorDetail[];
  }
>(
  id: string,
  data: P,
) => {
  const res = await fetch(`${URL}/api/categories/${id}`, {
    method: 'PUT',
    credentials: 'same-origin',
    headers: {
      'Content-type': 'application/json',
    },
    body: JSON.stringify(data),
  });

  return res.json() as Promise<R>;
};

export const deleteCategory = async <R extends ApiResponse>(id: string) => {
  const res = await fetch(`${URL}/api/categories/${id}`, {
    method: 'DELETE',
    credentials: 'same-origin',
  });

  return res.json() as Promise<R>;
};

export const createProduct = async <
  R extends ApiResponse & { product: Product; errors?: ProductErrorDetail[] }
>(
  data: FormData,
) => {
  const res = await fetch(`${URL}/api/products/create`, {
    method: 'POST',
    credentials: 'same-origin',
    body: data,
  });

  return res.json() as Promise<R>;
};

export const updateProduct = async <
  R extends ApiResponse & { product: Product; errors?: ProductErrorDetail[] }
>(
  id: string,
  data: FormData,
) => {
  const res = await fetch(`${URL}/api/products/update/${id}`, {
    method: 'PUT',
    credentials: 'same-origin',
    body: data,
  });

  return res.json() as Promise<R>;
};

export const deleteProduct = async <R extends ApiResponse>(id: string) => {
  const res = await fetch(`${URL}/api/products/delete/${id}`, {
    method: 'DELETE',
    credentials: 'same-origin',
  });

  return res.json() as Promise<R>;
};

export const findProduct = async <
  R extends ApiResponse & {
    product: Product;
    errors?: ProductNameErrorDetail[];
  }
>({
  name,
}: ProductName) => {
  const res = await fetch(`${URL}/api/products?search=${name}`);

  return res.json() as Promise<R>;
};

export const getProducts = async <
  P extends string[] | null,
  R extends ApiResponse & { products: Product[]; count: number }
>(
  filters: P,
  limit = 3,
  offset = 0,
) => {
  const res = await fetch(
    `${URL}/api/products${
      filters ? `?filters=${filters.join(' ')}` : ''
    }&limit=${limit}&offset=${offset}`,
  );

  return res.json() as Promise<R>;
};

export const getProductsByCategory = async <
  P extends string[] | null,
  R extends ApiResponse & { products: Product[]; count: number }
>(
  name: string,
  filters: P,
  limit = 3,
  offset = 0,
) => {
  const res = await fetch(
    `${URL}/api/products?search=${name}&byCategory=true${
      filters ? `&filters=${filters.join(' ')}` : ''
    }&limit=${limit}&offset=${offset}`,
  );

  return res.json() as Promise<R>;
};

export const createOrder = async <
  P extends OrderFields,
  R extends ApiResponse & { number: string }
>(
  data: P,
) => {
  const res = await fetch(`${URL}/api/orders/create`, {
    method: 'POST',
    credentials: 'same-origin',
    headers: {
      'Content-type': 'application/json',
    },
    body: JSON.stringify(data),
  });

  return res.json() as Promise<R>;
};

export const findOrder = async <
  R extends ApiResponse & { order: Order; errors?: OrderNumberErrorDetail[] }
>({
  number,
}: OrderNumber) => {
  const res = await fetch(`${URL}/api/orders?search=${number}`, {
    credentials: 'same-origin',
  });

  return res.json() as Promise<R>;
};

export const updateOrder = async <
  P extends OrderStatus,
  R extends ApiResponse & { order: Order; errors?: OrderStatusErrorDetail[] }
>(
  id: string,
  data: P,
) => {
  const res = await fetch(`${URL}/api/orders/update/${id}`, {
    credentials: 'same-origin',
    method: 'PUT',
    headers: {
      'Content-type': 'application/json',
    },
    body: JSON.stringify(data),
  });

  return res.json() as Promise<R>;
};

export const deleteOrder = async <R extends ApiResponse>(id: string) => {
  const res = await fetch(`${URL}/api/orders/delete/${id}`, {
    method: 'DELETE',
    credentials: 'same-origin',
  });

  return res.json() as Promise<R>;
};

export const getOrders = async <
  P extends string[] | null,
  R extends ApiResponse & ManagementOrders
>(
  filters: P,
  limit = 3,
  offset = 0,
) => {
  const res = await fetch(
    `${URL}/api/orders${
      filters ? `?filters=${filters.join(' ')}` : ''
    }&limit=${limit}&offset=${offset}`,
    { credentials: 'same-origin' },
  );

  return res.json() as Promise<R>;
};

export const getUserOrders = async <
  P extends string[] | null,
  R extends ApiResponse & UserOrders
>(
  filters: P,
  limit = 3,
  offset = 0,
) => {
  const res = await fetch(
    `${URL}/api/orders${
      filters ? `?filters=${filters.join(' ')}` : ''
    }&limit=${limit}&offset=${offset}&byUserId=true`,
    { credentials: 'same-origin' },
  );

  return res.json() as Promise<R>;
};

export const getRecommendedProducts = async <
  R extends ApiResponse & { products: ProductWithCategory[] }
>(
  limit = 3,
  offset = 0,
) => {
  const res = await fetch(
    `${URL}/api/products?recommended=true&limit=${limit}&offset=${offset}`,
  );

  return res.json() as Promise<R>;
};
