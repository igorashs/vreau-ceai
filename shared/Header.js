import styled from 'styled-components';
import Link from 'next/link';
import Image from 'next/image';
import { IconButton } from './IconButton';
import { Container } from './Container';
import MenuSvg from '@icons/menu.svg';
import CartSvg from '@icons/cart.svg';
import LoginSvg from '@icons/login.svg';
import breakpoints from 'GlobalStyle/breakpoints';
import { useState } from 'react';

const navLinks = [
  {
    path: '/categories',
    text: 'Ceai',
    label: 'Deschideți pagina cu categorii',
    Icon: null,
    accent: false
  },

  {
    path: '/about',
    text: 'Despre',
    label: 'Deschideți pagina cu informații despre noi',
    Icon: null,
    accent: false
  },
  {
    path: '/contacts',
    text: 'Contacte',
    label: 'Deschideți pagina cu contacte',
    Icon: null,
    accent: false
  },

  {
    path: '/cart',
    text: 'Coș',
    label: 'Deschideți pagina cu coșul personal',
    Icon: CartSvg,
    accent: false
  },
  {
    path: '/login',
    text: 'Conectați-vă',
    label: 'Deschideți pagina pentru conectare',
    Icon: LoginSvg,
    accent: true
  }
];

const Wrapper = styled.div`
  display: flex;
  align-items: center;
  background-color: var(--layout-dark);
`;

const StyledHeader = styled.header`
  display: grid;
  grid-template-columns: auto 1fr;
  grid-template-rows: calc(var(--baseline) * 2) auto;
  align-items: center;
  justify-items: end;
`;

const Nav = styled.nav`
  width: 100%;
  justify-self: start;
  grid-column: span 2;
  padding-bottom: calc(var(--baseline) / 2);

  @media (min-width: ${breakpoints.lg}) {
    display: initial;
    grid-column: 2;
    grid-row: 1;
    padding: 0;
    border: 0;
  }

  ${({ hide }) => hide && 'display: none'}
`;

const NavList = styled.ul`
  display: flex;
  flex-direction: column;
  border-bottom: 1px solid var(--layout);

  @media (min-width: ${breakpoints.lg}) {
    flex-direction: row;
    justify-content: flex-end;
    gap: var(--baseline);
    border: 0;
  }
`;

const NavListItem = styled.li`
  border-top: 1px solid var(--layout);
  padding: 7px 0;

  a {
    color: var(--text-light);
    text-decoration: none;
    display: flex;
    align-items: center;
    gap: 7px;
  }

  @media (min-width: ${breakpoints.lg}) {
    border: 0;
  }

  ${({ accent }) => accent && ' a {color: var(--accent-text-light)}'}
`;

const Logo = styled.div`
  justify-self: start;
  display: flex;
  height: 32px;
  min-width: 158px;
`;

const NavToggle = styled.div`
  @media (min-width: ${breakpoints.lg}) {
    display: none;
  }
`;

export function Header() {
  const [isHidden, setIsHidden] = useState(true);

  return (
    <Wrapper>
      <Container>
        <StyledHeader>
          <Logo>
            <Link href="/">
              <a aria-label="Acasă">
                <Image
                  src="/logo.svg"
                  alt="vreau ceai logo"
                  quality="100"
                  width="158px"
                  height="32px"
                />
              </a>
            </Link>
          </Logo>
          <NavToggle>
            <IconButton
              aria-label="Comutați Navigarea"
              onClick={() => setIsHidden((prev) => !prev)}
            >
              <MenuSvg />
            </IconButton>
          </NavToggle>
          <Nav hide={isHidden}>
            <NavList>
              {navLinks.map(({ path, text, label, Icon, accent }) => (
                <NavListItem key={path} accent={accent}>
                  <Link href={path}>
                    <a aria-label={label}>
                      {Icon && <Icon />}
                      {text}
                    </a>
                  </Link>
                </NavListItem>
              ))}
            </NavList>
          </Nav>
        </StyledHeader>
      </Container>
    </Wrapper>
  );
}
