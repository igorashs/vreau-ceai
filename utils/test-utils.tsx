import { render, RenderOptions } from '@testing-library/react';
import GlobalStyle from 'GlobalStyle';

const Wrapper: React.FC = ({ children }) => (
  <>
    <GlobalStyle />
    {children}
  </>
);

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
) => render(ui, { wrapper: Wrapper, ...options });

export * from '@testing-library/react';

export { customRender as render };
