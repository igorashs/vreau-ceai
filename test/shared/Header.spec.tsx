import { fireEvent, renderWithSession, waitFor } from '@/utils/test-utils';
import Header from '@/shared/Header';
import { logout } from 'services/ceaiApi';
import { useRouter } from 'next/router';
import { loginLink, managementLink, myOrdersLink } from '@/utils/links';

const user = {
  _id: 'user1',
  name: 'username',
  isAdmin: false,
  isManager: false,
};

const session = {
  isAuth: false,
  user: null,
  needRefresh: false,
};

jest.mock('next/router', () => ({
  useRouter: jest.fn(),
}));

jest.mock('services/ceaiApi', () => ({
  logout: jest
    .fn()
    .mockImplementationOnce(() =>
      Promise.resolve({ success: true, message: 'Success' }),
    )
    .mockImplementationOnce(() =>
      Promise.resolve({ success: false, message: 'Bad Request' }),
    ),
}));

describe('Header', () => {
  beforeEach((logout as jest.Mock).mockClear);

  it('renders correctly', () => {
    const {
      container,
      getByText,
      queryByText,
      getByTestId,
      queryByTestId,
    } = renderWithSession(<Header />, session);

    expect(getByText(loginLink.text)).not.toBeVisible();
    expect(queryByText(managementLink.text)).not.toBeInTheDocument();
    expect(queryByTestId('logout-btn')).not.toBeInTheDocument();
    expect(getByTestId('toggle-btn').parentElement).toHaveStyleRule(
      'display',
      'none',
      {
        media: '(min-width:992px)',
      },
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  describe('.onNavToggleClick', () => {
    it('toggles nav list', () => {
      const { getByTestId, getByText } = renderWithSession(<Header />, session);

      const toggleBtn = getByTestId('toggle-btn');
      fireEvent.click(toggleBtn);

      expect(getByText(loginLink.text)).toBeVisible();
      fireEvent.click(toggleBtn);
      expect(getByText(loginLink.text)).not.toBeVisible();
    });
  });

  describe('when user is authenticated', () => {
    describe('when user is updated', () => {
      it('renders with the correct links', () => {
        const {
          getByText,
          queryByText,
          rerenderWithSession,
        } = renderWithSession(<Header />, {
          ...session,
          isAuth: true,
          user,
        });

        expect(queryByText(managementLink.text)).not.toBeInTheDocument();
        expect(getByText(myOrdersLink.text)).toBeInTheDocument();
        expect(getByText(user.name)).toBeInTheDocument();

        rerenderWithSession(<Header />, {
          ...session,
          isAuth: true,
          user: { ...user, isManager: true },
        });

        expect(getByText(managementLink.text)).toBeInTheDocument();
      });
    });

    describe('.onLogoutClick', () => {
      it('reloads with success', async () => {
        const mockRouter = {
          reload: jest.fn(),
        };

        (useRouter as jest.Mock).mockReturnValue(mockRouter);

        const { getByTestId } = renderWithSession(<Header />, {
          ...session,
          isAuth: true,
          user,
        });

        fireEvent.click(getByTestId('logout-btn'));

        await waitFor(() => expect(logout).toBeCalledTimes(1));
        expect(mockRouter.reload).toBeCalledTimes(1);
      });

      it("doesn't reload", async () => {
        const mockRouter = {
          reload: jest.fn(),
        };

        (useRouter as jest.Mock).mockReturnValue(mockRouter);

        const { getByTestId } = renderWithSession(<Header />, {
          ...session,
          isAuth: true,
          user,
        });

        fireEvent.click(getByTestId('logout-btn'));

        expect(logout).toBeCalledTimes(1);
        expect(mockRouter.reload).toBeCalledTimes(0);
      });
    });

    describe('when user is a client', () => {
      it('renders user data with private links', () => {
        const { getByText, queryByText } = renderWithSession(<Header />, {
          ...session,
          isAuth: true,
          user,
        });

        expect(queryByText(loginLink.text)).not.toBeInTheDocument();
        expect(getByText(myOrdersLink.text)).toBeInTheDocument();
        expect(getByText(user.name)).toBeInTheDocument();
      });
    });

    describe('when user is a manager/admin', () => {
      it('renders management link', () => {
        const { getByText, rerenderWithSession } = renderWithSession(
          <Header />,
          {
            ...session,
            isAuth: true,
            user: { ...user, isManager: true },
          },
        );

        expect(getByText(managementLink.text)).toBeInTheDocument();

        rerenderWithSession(<Header />, {
          ...session,
          isAuth: true,
          user: { ...user, isAdmin: true },
        });

        expect(getByText(managementLink.text)).toBeInTheDocument();
      });
    });
  });
});
