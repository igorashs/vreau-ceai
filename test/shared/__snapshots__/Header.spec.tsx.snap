// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
.c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 7px;
  padding: calc(var(--baseline) / 4) calc(var(--baseline) / 2);
  border-radius: 4px;
  border: 0;
  color: var(--text-light);
  background-color: var(--accent-dark);
  box-shadow: 1px 1px 2px #00000066;
  text-transform: lowercase;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  background-color: transparent;
  box-shadow: none;
  padding: 4px;
  box-shadow: none;
}

.c5:hover {
  background-color: var(--accent);
}

.c5:hover {
  background-color: #0000001a;
}

.c9 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 7px;
  color: var(--text-light);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c9:hover {
  -webkit-filter: brightness(0.7);
  filter: brightness(0.7);
}

.c10 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 7px;
  color: var(--accent-text-light);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c10:hover {
  -webkit-filter: brightness(0.7);
  filter: brightness(0.7);
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: var(--layout-dark);
}

.c2 {
  display: grid;
  -webkit-column-gap: var(--baseline);
  column-gap: var(--baseline);
  grid-template-columns: auto 1fr;
  grid-template-rows: calc(var(--baseline) * 2) auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  justify-items: end;
}

.c6 {
  width: 100%;
  justify-self: start;
  grid-column: span 2;
  padding-bottom: calc(var(--baseline) / 2);
  display: none;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border-bottom: 1px solid var(--layout);
}

.c8 {
  border-top: 1px solid var(--layout);
  padding: 7px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: calc(var(--baseline) * 2);
}

.c3 {
  justify-self: start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 32px;
  min-width: 158px;
}

@media (min-width:576px) {
  .c1 {
    width: 540px;
    max-width: 540px;
    margin: 0 auto;
  }
}

@media (min-width:768px) {
  .c1 {
    width: 720px;
    max-width: 720px;
  }
}

@media (min-width:992px) {
  .c1 {
    width: 960px;
    max-width: 960px;
  }
}

@media (min-width:1200px) {
  .c1 {
    width: 1140px;
    max-width: 1140px;
  }
}

@media (max-width:576px) {
  .c1 {
    width: 100%;
    padding: 0 15px;
  }
}

@media (min-width:992px) {
  .c6 {
    display: initial;
    grid-column: 2;
    grid-row: span 2;
    padding: 0;
    border: 0;
  }
}

@media (min-width:992px) {
  .c7 {
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    -webkit-column-gap: var(--baseline);
    column-gap: var(--baseline);
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-pack: end;
    -webkit-justify-content: flex-end;
    -ms-flex-pack: end;
    justify-content: flex-end;
    border: 0;
  }
}

@media (min-width:992px) {
  .c8 {
    border: 0;
  }
}

@media (min-width:992px) {
  .c4 {
    display: none;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <header
      class="c2"
    >
      <div
        class="c3"
      >
        <a
          aria-label="Acasă"
          href="/"
        >
          <div
            style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
          >
            <div
              style="box-sizing: border-box; display: block; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                role="presentation"
                src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTU4IiBoZWlnaHQ9IjMyIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZlcnNpb249IjEuMSIvPg=="
                style="max-width: 100%; display: block;"
              />
            </div>
            <img
              alt="vreau ceai logo"
              decoding="async"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </div>
        </a>
      </div>
      <div
        class="c4"
      >
        <button
          aria-label="Comutați Navigarea"
          class="c5"
          data-testid="toggle-btn"
        >
          <svg />
        </button>
      </div>
      <nav
        class="c6"
      >
        <ul
          class="c7"
        >
          <li
            class="c8"
          >
            <a
              aria-label="Deschideți pagina cu categorii"
              class="c9"
              href="/categories"
              rel=""
              target=""
            >
              Ceai
            </a>
          </li>
          <li
            class="c8"
          >
            <a
              aria-label="Deschideți pagina cu informații despre noi"
              class="c9"
              href="/about"
              rel=""
              target=""
            >
              Despre
            </a>
          </li>
          <li
            class="c8"
          >
            <a
              aria-label="Deschideți pagina cu contacte"
              class="c9"
              href="/contacts"
              rel=""
              target=""
            >
              Contacte
            </a>
          </li>
          <li
            class="c8"
          >
            <a
              aria-label="Deschideți pagina cu coșul personal"
              class="c9"
              href="/cart"
              rel=""
              target=""
            >
              <svg />
              Coș
            </a>
          </li>
          <li
            class="c8"
          >
            <a
              aria-label="Deschideți pagina pentru conectare"
              class="c10"
              href="/login"
              rel=""
              target=""
            >
              <svg />
              Conectați-vă
            </a>
          </li>
        </ul>
      </nav>
    </header>
  </div>
</div>
`;
